### PR TITLE
hook refinements: second pass

### PR DESCRIPTION
**Pull request by le-jeu**
This follows #250, reversing the logic caller/callee with the benefit of some factorization.

Event | Description
-- | --
wasabee:agentlocations | when we should have/get new teammate locations
wasabee:crosslinks | to run crosslinks
wasabee:crosslinks:done | when crosslink is done
wasabee:defensivekeys | when we should have new defensive keys
wasabee:fullsync | on full sync
wasabee:login | on login
wasabee:logout | on logout/purge
wasabee:op:background | on background map toggle
wasabee:op:change | on operation (current) changes
wasabee:op:delete | on operation deletion
wasabee:op:select | when the user selects another operation
wasabee:op:showhide | on operation visibility change (in ops list)
wasabee:pane:clear | on dialog close (when a pane, used to clear and close the pane)
wasabee:pane:set | on dialog open (when a pane, used to set wasabee pane content)
wasabee:portal:click | on portal click (bypass IITC portalSelected hook)
wasabee:teams | on teams update
wasabee:team:update | on team member update
wasabee:ui:lang | on lang change
wasabee:ui:skin | on skin change

Additional feature:
 - Sync loading animation
 - Trust the server on logout, even if it's wrong
 - OpsList from IDB

### CODE REVIEW

File: src/code/buttons/quickdrawButton.js [hunk 1]
```diff
@@ -113,7 +113,6 @@ const QuickDrawControl = L.Handler.extend({
   },
 
   addHooks: function () {
-    if (!this._map) return;
```
Comment (cloudkucooland) [inline src/code/buttons/quickdrawButton.js:116]: yes, we only ever use window.map

File: src/code/dialogs/blockersList.js [hunk 2]
```diff
@@ -31,8 +31,8 @@ const BlockerList = WDialog.extend({
 
   removeHooks: function () {
     WDialog.prototype.removeHooks.call(this);
-    window.map.off("wasabee:uiupdate:mapdata", this.blockerlistUpdate, this);
-    window.map.off("wasabee:crosslinks:done", this.blockerlistUpdate, this);
+    window.map.off("wasabee:op:select wasabee:op:change", this.update, this);
```
Comment (cloudkucooland) [inline src/code/dialogs/blockersList.js:34]: does this work? space separated list?

**Review by cloudkucooland — APPROVED** [review]
This all looks fine. I'm not sure what you mean by "reversing the logic" but this seems sane.

**Comment by cloudkucooland** [comment]
can you add "wasabee:op:zone" for when the zone list or polygons change?

File: src/code/dialogs/blockersList.js (hunk 2 above)
Comment (le-jeu) [reply src/code/dialogs/blockersList.js:34]: yes: https://leafletjs.com/reference-1.7.1.html#evented-on
> You can also pass several space-separated types (e.g. 'click dblclick').

**Comment by le-jeu** [comment]
how and where would you use them ? `wasabee:op:change` covers all changes on the current op, including zones (it's basically fired on any .update (thus by hand after .store)

File: src/code/selectedOp.js [hunk 3]
```diff
@@ -192,25 +192,23 @@ export async function setOpBackground(opID, background) {
 
 export async function opsList(hidden = true) {
   // after 0.19, remove the list and just query the idb keys
```
Comment (cloudkucooland) [inline src/code/selectedOp.js:194]: per this comment, do we still need to maintain the list?

File: src/code/selectedOp.js [hunk 4]
```diff
@@ -192,25 +192,23 @@ export async function setOpBackground(opID, background) {
...
-    } catch (e) {
-      console.error(e);
-    }
+  let ops = [];
+  try {
+    const raw =
+      localStorage[window.plugin.wasabee.static.constants.OPS_LIST_KEY];
+    ops = JSON.parse(raw);
+  } catch {
+    //
```
Comment (cloudkucooland) [inline src/code/selectedOp.js:201]: empty catch? do we need to log something?

**Review by cloudkucooland — APPROVED** [review]

File: src/code/uiCommands.js [hunk 5]
```diff
@@ -449,13 +449,14 @@ export async function fullSync() {
     else if (!ol.includes(so.ID)) await changeOpIfNeeded();
     // update UI to reflect new ops list
     // XXX do we need a specific call for "op list update"?
```
Comment (cloudkucooland) [inline src/code/uiCommands.js:451]: if fullsync covers it, we can remove this XXX comment

File: src/code/selectedOp.js (hunk 4 above)
Comment (le-jeu) [reply src/code/selectedOp.js:201]: faulty data in `localStorage[window.plugin.wasabee.static.constants.OPS_LIST_KEY];`
nothing to rely on so nothing to log

File: src/code/selectedOp.js (hunk 3 above)
Comment (le-jeu) [reply src/code/selectedOp.js:194]: that's only to catch pre 0.19 local ops

**Pull request merged**